app/uk/gov/hmrc/cdsreimbursementclaim/controllers/SubmitClaimController.scala

```diff
@@ -16,40 +16,149 @@
 import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging
 import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging.LoggerOps
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.workitem.WorkItem
 
-import javax.inject.{Inject, Singleton}
-import scala.concurrent.ExecutionContext
-
-final case class SubmitClaimResponse(
-  caseNumber: String,
-  payService: String,
-  processingDate: String
-)
-object SubmitClaimResponse {
-  implicit val format: OFormat[SubmitClaimResponse] = Json.format[SubmitClaimResponse]
-}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
 
 @Singleton()
-class SubmitClaimController @Inject() (eisService: SubmitClaimService, cc: ControllerComponents)(implicit
+class SubmitClaimController @Inject() (
+  authenticate: AuthenticateActions,
+  eisService: SubmitClaimService,
+  upscanService: UpscanService,
+  fileUploadQueue: FileUploadQueue,
+  cc: ControllerComponents
+)(implicit
   ec: ExecutionContext
 ) extends BackendController(cc)
     with Logging {
 
-  def claim(): Action[JsValue] = Action.async(parse.json) { implicit request =>
-    eisService
-      .submitClaim(request.body)
+  def claim(): Action[JsValue] = authenticate(parse.json).async { implicit request =>
+    (for {
+      frontendClaim <- EitherT.fromEither[Future](Try(request.body.as[FrontendSubmitClaim]).toEither.leftMap(Error(_)))
+      claimResponse <- eisService.submitClaim(createSubmitClaim(frontendClaim))
+      _             <- createAndStoreDec64Requests(frontendClaim, claimResponse)
+    } yield claimResponse)
       .fold(
         e => {
           logger.warn(s"could not submit claim", e)
           InternalServerError
         },
-        response => Ok(response)
+        response => Ok(Json.toJson(response))
       )
   }
+
+  def createSubmitClaim(frontendSubmitClaim: FrontendSubmitClaim): SubmitClaimRequest = {
+    val requestCommon        = RequestCommon()
+    val requestDetail        = frontendSubmitClaim.declarationIdType match {
+      case MRNType   =>
+        val mrnDetails = MrnDetails(frontendSubmitClaim.declarationId)
+        RequestDetail(frontendSubmitClaim.eori).copy(MRNDetails = Some(List(mrnDetails)))
+      case CHIEFType =>
+        val entryDetails = EntryDetails(frontendSubmitClaim.declarationId)
+        RequestDetail(frontendSubmitClaim.eori).copy(entryDetails = Some(List(entryDetails)))
+    }
+    val postNewClaimsRequest = PostNewClaimsRequest(requestCommon, requestDetail)
+    SubmitClaimRequest(postNewClaimsRequest)
+  }
+
+  def createAndStoreDec64Requests(claimRequest: FrontendSubmitClaim, claimResponse: SubmitClaimResponse)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, Error, List[WorkItem[WorkItemPayload]]] = {
+    val uploadReferences = claimRequest.files.map(_.uploadReference)
+    for {
+      uploads  <- upscanService.readUpscanUploads(uploadReferences)
```

Comment:
This is unnecessary. As we discussed a few times this information will already been present and passed to the back-end.

Reply:
The whole class depends on this first line to get the files.
When the frontend - backend message contains this, I will refactor it.